packages/toolpad-core/src/SignInPage/SignInPage.tsx

```diff
@@ -239,388 +250,421 @@ function SignInPage(props: SignInPageProps) {
+                        setFormStatus((prev) => ({
+                          ...prev,
+                          loading: oauthResponse?.error || docs ? false : prev.loading,
+                          error: oauthResponse?.error,
+                        }));
+                      }}
+                    >
+                      <LoadingButton
+                        key={provider.id}
+                        variant="outlined"
+                        type="submit"
+                        fullWidth
+                        size="large"
+                        disableElevation
+                        name="provider"
+                        color="inherit"
+                        loading={loading && selectedProviderId === provider.id}
+                        value={provider.id}
+                        startIcon={IconProviderMap.get(provider.id)}
+                        sx={{
+                          textTransform: 'capitalize',
+                        }}
+                      >
+                        <span>Sign in with {provider.name}</span>
+                      </LoadingButton>
+                    </form>
+                  );
+                })}
+            </Stack>
+
+            {passkeyProvider ? (
+              <React.Fragment>
+                {singleProvider ? null : <Divider sx={{ mt: 2, mx: 0, mb: 1 }}>or</Divider>}
+                {error && selectedProviderId === 'passkey' ? (
+                  <Alert sx={{ my: 2 }} severity="error">
+                    {error}
+                  </Alert>
+                ) : null}
+                <Box
+                  component="form"
+                  onSubmit={async (event) => {
+                    setFormStatus({
+                      error: '',
+                      selectedProviderId: passkeyProvider.id,
+                      loading: true,
+                    });
+                    event.preventDefault();
+                    const formData = new FormData(event.currentTarget);
+                    const passkeyResponse = await signIn?.(passkeyProvider, formData, callbackUrl);
+                    setFormStatus((prev) => ({
+                      ...prev,
+                      loading: false,
+                      error: passkeyResponse?.error,
+                    }));
+                  }}
+                >
+                  {slots?.emailField ? (
+                    <slots.emailField {...slotProps?.emailField} />
+                  ) : (
+                    <Box sx={{ marginBottom: (theme) => theme.spacing(1) }}>
+                      <InputLabel shrink htmlFor="email-passkey" sx={{ marginBottom: 0 }}>
+                        Email
+                      </InputLabel>
+                      <TextField
+                        required
+                        slotProps={{
+                          htmlInput: {
+                            sx: (theme) => ({
```

Comment:
Also if we're using the theme so much, we could just call it from the `useTheme` hook?